frappe/database/database.py

```diff
@@ -32,13 +32,15 @@
 from frappe.query_builder.functions import Count
 from frappe.utils import cast as cast_fieldtype
 from frappe.utils import cint, get_datetime, get_table_name, getdate, now, sbool
-from frappe.utils.deprecations import deprecated, deprecation_warning
+from frappe.utils.deprecations import deprecation_warning
 
 IFNULL_PATTERN = re.compile(r"ifnull\(", flags=re.IGNORECASE)
 INDEX_PATTERN = re.compile(r"\s*\([^)]+\)\s*")
 SINGLE_WORD_PATTERN = re.compile(r'([`"]?)(tab([A-Z]\w+))\1')
 MULTI_WORD_PATTERN = re.compile(r'([`"])(tab([A-Z]\w+)( [A-Z]\w+)+)\1')
 
+SQL_ITERATOR_BATCH_SIZE = 100
```

Comment:
This should be configurable per query? :thinking: 


Basically sacrificing memory usage for reduced # of network calls.

Reply:
I can see how report like queries can use it. Assuming reports can tolerate up to 100MB of additional memory usage the ideal batch size would be much much higher (and changing based on query/return types)